src/shared/switch-root.c

```diff
@@ -54,11 +54,11 @@ int switch_root(const char *new_root,
         assert(new_root);
 
         /* Check if we shall remove the contents of the old root */
-        old_root_fd = open("/", O_DIRECTORY|O_CLOEXEC);
+        old_root_fd = open("/", O_PATH|O_CLOEXEC);
```

Comment:
hmm, are you sure this works? we do rm_rf_children() later, where we enumerate the dentries

Reply:
dropping